ng/src/web/pages/portlists/dialog.js

```diff
@@ -65,95 +73,108 @@ const PortListsDialog = ({
+
   return (
-    <Layout flex="column">
-
-      <FormGroup title={_('Name')}>
-        <TextField
-          name="name"
-          value={name}
-          grow="1"
-          size="30"
-          onChange={onValueChange}
-          maxLength="80"/>
-      </FormGroup>
-
-      <FormGroup title={_('Comment')}>
-        <TextField
-          name="comment"
-          value={comment}
-          grow="1"
-          size="30"
-          maxLength="400"
-          onChange={onValueChange}/>
-      </FormGroup>
-
-      {!is_edit &&
-        <FormGroup title={_('Port Ranges')} flex="column">
-          <Divider flex="column">
-            <Divider>
-              <Radio
-                title={_('Manual')}
-                name="from_file"
-                value="0"
-                onChange={onValueChange}
-                checked={from_file !== '1'}/>
+    <SaveDialog
+      visible={visible}
+      title={title}
+      onClose={onClose}
+      onSave={onSave}
+      initialData={data}
+    >
+      {({
+        data: state,
+        onValueChange,
+      }) => {
+        return (
+          <Layout flex="column">
+
+            <FormGroup title={_('Name')}>
               <TextField
+                name="name"
+                value={state.name}
                 grow="1"
-                name="port_range"
-                value={port_range}
-                disabled={from_file === '1'}
+                size="30"
                 onChange={onValueChange}
-                size="30" maxLength="400"/>
-            </Divider>
-            <Divider>
-              <Radio
-                title={_('From file')}
-                name="from_file"
-                value="1"
-                onChange={onValueChange}
-                checked={from_file === '1'}/>
-              <FileField
-                name="file"
+                maxLength="80"/>
+            </FormGroup>
+
+            <FormGroup title={_('Comment')}>
+              <TextField
+                name="comment"
+                value={state.comment}
+                grow="1"
+                size="30"
+                maxLength="400"
                 onChange={onValueChange}/>
-            </Divider>
-          </Divider>
-        </FormGroup>
-      }
-      {is_edit &&
-        <Section title={_('Port Ranges')} extra={newrangeicon}>
-          {is_defined(port_list) &&
-            <PortRangesTable
-              portRanges={port_list.port_ranges}
-              onDeleteClick={onDeletePortRangeClick}/>
-          }
-        </Section>
-      }
-    </Layout>
+            </FormGroup>
+
+            {!is_edit &&
+              <FormGroup title={_('Port Ranges')} flex="column">
+                <Divider flex="column">
+                  <Divider>
+                    <Radio
+                      title={_('Manual')}
+                      name="from_file"
+                      value="0"
+                      onChange={onValueChange}
+                      checked={from_file !== '1'}/>
```

Comment:
Should use NO_VALUE and YES_VALUE. Maybe setting convert={parse_yesno} is necessary too. I am not sure yet.